### PR TITLE
SG-30560 Code cleanup for qt4 and pyside

### PR DESCRIPTION
**Pull request by eduardoChaucaGallegos**
Code cleanup for qt4 and pyside

### CODE REVIEW

File: python/app/qt_importer.py [hunk 1]
```diff
@@ -41,11 +41,8 @@
         except ImportError:
             pass
 
-    if not imported_qt:
-        # Try PySide
-        from PySide import QtCore, QtGui
```
Comment (julien-lang) [inline python/app/qt_importer.py:46]: No, please remove the except ImportError: from PySide6 above to get the same behavior if we can't find a PySide interface

```python
    if not imported_qt:
        # Try PySide6
         from PySide6 import QtCore
         import PySide6.QtGui as __PySide_QtGui
         import PySide6.QtWidgets as __PySide_QtWidgets
    else:
        # Patch PySide2/PySide6 to keep the QtGui interface consi
        [...]
```

File: python/app/ui/resources_rc.py [hunk 2]
```diff
@@ -15,7 +15,7 @@
         try:
             from PySide6 import QtCore
         except ImportError:
-            from PySide import QtCore
+            pass
```
Comment (julien-lang) [inline python/app/ui/resources_rc.py]: No, we want an ImportError raise eventually:

```python
try:
    from sgtk.platform.qt import QtCore
except ImportError:
    try:
        from PySide2 import QtCore
    except ImportError:
        from PySide6 import QtCore
```

**Review by julien-lang — CHANGES_REQUESTED** [review]

**Review by julien-lang — APPROVED** [review]
LGTM!

File: python/app/qt_importer.py [hunk 3]
```diff
@@ -32,20 +32,12 @@
...
 
-            imported_qt = True
-        except ImportError:
-            pass
-
-    if not imported_qt:
-        # Try PySide
-        from PySide import QtCore, QtGui
     else:
-        # Patch PySide2/PySide6 to keep the QtGui interface consistent with PySide (Qt 4).
```
Comment (carlos-villavicencio-adsk) [inline python/app/qt_importer.py:48]: We still use the Qt interface consistent with Qt4. Please revert this comment.

File: python/app/qt_importer.py [hunk 4]
```diff
@@ -68,10 +60,5 @@ def _move_attributes(dst, src, names):
...
 # Handle QRegExp / QRegularExpression
-if hasattr(QtCore, "QRegularExpression"):
-    qt_re_module = QtCore.QRegularExpression
-    qt_re_module_is_regular_expression = True
-else:
-    # NOTE remove once Qt4 support is removed
-    qt_re_module = QtCore.QRegExp
-    qt_re_module_is_regular_expression = False
+qt_re_module = QtCore.QRegularExpression
+qt_re_module_is_regular_expression = True
```
Comment (carlos-villavicencio-adsk) [inline python/app/qt_importer.py]: Unfortunately, this requires more work on python/app/syntax_highlighter.py.

You can take a look at #25. Probably it's worth removing dead code such as `match_multiline_regexp` and `highlight_block_regexp`.

**Review by carlos-villavicencio-adsk — CHANGES_REQUESTED** [review]

**Review by carlos-villavicencio-adsk — APPROVED** [review]
LGTM! Thanks

**Pull request merged**